packages/babel-helpers/src/helpers.js

```diff
@@ -399,23 +399,34 @@ helpers.inherits = () => template.program.ast`
     if (typeof superClass !== "function" && superClass !== null) {
       throw new TypeError("Super expression must either be null or a function");
     }
-    setPrototypeOf(subClass.prototype, superClass && superClass.prototype);
+    subClass.prototype = Object.create(superClass && superClass.prototype, {
+      constructor: {
+        value: subClass,
+        writable: true,
+        configurable: true
+      }
+    });
     if (superClass) setPrototypeOf(subClass, superClass);
   }
 `;
 
 helpers.inheritsLoose = () => template.program.ast`
   export default function _inheritsLoose(subClass, superClass) {
-    subClass.prototype.__proto__ = superClass && superClass.prototype;
+    subClass.prototype = Object.create(superClass.prototype);
+    subClass.prototype.constructor = subClass;
     subClass.__proto__ = superClass;
   }
 `;
 
 helpers.getPrototypeOf = () => template.program.ast`
   export default function _getPrototypeOf(o) {
-    _getPrototypeOf = Object.getPrototypeOf || function _getPrototypeOf(o) {
-      return o.__proto__;
-    };
+    if (Object.setPrototypeOf && Object.getPrototypeOf) {
+      _getPrototypeOf = Object.getPrototypeOf;
+    } else {
+      _getPrototypeOf = function _getPrototypeOf(o) {
+        return o.__proto__;
```

Comment:
I think my comment here was misinterpreted. My concern was that this does not preserve the behavior changed in https://github.com/babel/babel/pull/7675/files#diff-a6fddd06921a2b0da9eb1fb68cd794b9L14

Old super calls used to be `(Foo.__proto__ || Object.getPrototypeOf(Foo)).call`, but here we assume that `__proto__` is the source of truth and will always have a value, instead of falling back to `Object.getPrototypeOf` of `__proto__` is falsy.

Reply:
Wow, that was dyslexic of me. Fixed.